### PR TITLE
Add zerodayflag android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/35628

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by deeppandya** [comment]
CC @timchilds

File: android/java/org/chromium/base/BraveFeatureList.java [hunk 1]
```diff
@@ -31,4 +31,5 @@ public abstract class BraveFeatureList {
     public static final String AI_CHAT = "AIChat";
     public static final String BRAVE_SHOW_STRICT_FINGERPRINTING_MODE =
             "BraveShowStrictFingerprintingMode";
+    public static final String BRAVE_ZERO_DAY_FLAG_ANDROID = "brave-zero-day-flag-android";
```
Comment (samartnik) [inline android/java/org/chromium/base/BraveFeatureList.java:34]: We should probably use camel case style to follow Chromium's style they use in `ChromeFeatureList.java`. Also is this variable used anywhere?

Comment (deeppandya) [reply android/java/org/chromium/base/BraveFeatureList.java:34]: it's not being used right now but we will be using for product test soon.

**Review by samartnik — APPROVED** [review]
lgtm

**Review by iefremov — APPROVED** [review]
chromium_src

File: browser/about_flags.cc [hunk 2]
```diff
@@ -362,9 +362,18 @@
       kOsAndroid,                                                             \
       FEATURE_VALUE_TYPE(safe_browsing::features::kBraveAndroidSafeBrowsing), \
   })
+#define BRAVE_ZERO_DAY_FLAG_ANDROID                                        \
+  EXPAND_FEATURE_ENTRIES({                                                 \
+      "brave-zero-day-flag-android",                                       \
+      "ZeroDayFlag flag for product test",                                 \
+      "This flag will be set through griffin to perform product testing",  \
```
Comment (iefremov) [inline browser/about_flags.cc:369]: just ouf of curiousity, what sort of features would it cover? I.e. how is it different than normal per-feature flags ?

Comment (deeppandya) [reply browser/about_flags.cc:369]: @iefremov it would be mainly used for product A/B testing starting with new onboarding on android.

Comment (iefremov) [reply browser/about_flags.cc:369]: @deeppandya I understand, but any feature can be used for A/B testing, i.e. `kPlaylist` or something. I'm trying to understand an exact scenario of using of this flag (wondering b/c we try to keep an eye on what's changed via griffin for performance sheriffing purposes

Comment (deeppandya) [reply browser/about_flags.cc:369]: Sorry @iefremov for late response. i missed the comment here. Currently we don't have any feature associated with it. @timchilds can provide more details on why we need to add the flag.

**Pull request merged**